testers/markus_tester.py

```diff
@@ -360,7 +374,9 @@ def run(self):
                         else:
                             data_files = [data_files]
                         test = self.test_class(self, test_file, data_files, points, test_extra, feedback_open)
+                        self.before_test_run(test)
                         xml = test.run()
+                        self.after_test_run(test)
```

Comment:
I would recommend using a context manager or at least using a `try ... finally` block to ensure that `after_test_run` is executed even if an exception is raised in `test.run()`